src/Access/EnabledSettings.h

```diff
@@ -40,17 +41,23 @@ class EnabledSettings
     /// and the roles passed in the constructor.
     std::shared_ptr<const SettingsConstraints> getConstraints() const;
 
+    const std::vector<UUID> & getAppliedProfiles() const;
+
 private:
     friend class SettingsProfilesCache;
     EnabledSettings(const Params & params_);
 
-    void setSettingsAndConstraints(
-        const std::shared_ptr<const Settings> & settings_, const std::shared_ptr<const SettingsConstraints> & constraints_);
+    void setSettingsConstraintsAndProfiles(
```

Comment:
Let's insert one more `And` to make the name like `setSettingsAndConstraintsAndProfiles()`. Because `settings` and `settings constraints` are different things.

Reply:
done